Frameworks/Foundation/NSArray.mm

```diff
@@ -947,21 +923,68 @@ + (NSArray*)arrayWithContentsOfURL:(NSURL*)aURL {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self descriptionWithLocale:locale indent:0];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale indent:(NSUInteger)level {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSMutableString* s = [NSMutableString string];
+    NSString* indentStr = @"    ";
+    for (unsigned int i = 0; i < level; ++i) {
+        [s appendString:indentStr];
+    }
+
+    [s appendString:@"(\n"];
+
+    {
+        ++level;
+        auto deferPop = wil::ScopeExit([&level]() { --level; });
+        for (id val in self) {
+            for (unsigned int i = 0; i < level; ++i) {
```

Comment:
I'm still not sure this is totally correct; if you nest arrays, you'll double-indent them because you're appending `level*"spaces"`, and then passing level to the nested value which will prepend `level*"spaces"`